readthedocs/projects/forms.py

```diff
@@ -86,7 +86,7 @@ class ProjectBasicsForm(ProjectForm):
 
     class Meta:
         model = Project
-        fields = ('name', 'repo', 'repo_type', 'default_branch')
```

Comment:
This will also stop people from migrating off of old VCS to new right? I think we want to exclude it for projects using git, but keep it when they aren't, so they can change things? I think we also want to change the help_text to explain the situation to those projects.

Reply:
I often use a conditional pattern to change a form field's widget to HiddenInput whenever its irrelevant or vice-versa. This happens in the form's `__init__`.

Example:

```
class SomeForm(forms.ModelForm):
    dynamic_field = forms.BooleanField()
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        if self.instance and self.instance.blah = True:
            self.fields["dynamic_field"].widget = forms.HiddenInput()
```

By doing this, it's easier to phase out BZR/SVN/Mercurial support and still keep evolving the form for others. Hiding an input field in this way has much less impact on forms/views/models/templates than the alternatives.

Reply:
I updated this code to:

- don't allow importing new projects that are not Git
- allow changing repository type on those projects that are already imported